h/services/list_groups.py

```diff
@@ -51,6 +51,27 @@ def all_groups(self, user=None, authority=None, document_uri=None):
 
         return all_open_groups + private_groups
 
+    def associated_groups(self, authority, user=None):
+        """
+        Return a list of groups relevant to this session/profile (i.e. user).
+
+        Return a list of groups filtered on user and authority that the user
+        belongs to or is associated with (user is a creator/moderator/member).
+
+        If the user is None it returns and empty list.
+        """
+        # Return as quickly as possible. Since the user is unkown there will be no
+        # associated groups so might as well avoid the query.
+        if user is None:
+            return []
+        # Techically the call to open groups should return open and restricted groups,
+        # therefore check for membership as well as being a creator/moderator.
+        all_open_groups = [group for group in self._open_groups(user, authority)
```

Comment:
So "all_open_groups" is a list of what we are calling "open" groups _and_ a list of what we are calling "restricted" groups, which is a bit confusing.

We need to sort our terminology out and clarify what we use as a shorthand for "world-readable groups" (public?).